diff --git a/src/platform/common/externalinput.cc b/src/platform/common/externalinput.cc
index 96dc791c8320e7167689744c3df55b0de1578f7a..44e434406cb2e3d5c9685fd946a49ebed3e154ec 100644
--- a/src/platform/common/externalinput.cc
+++ b/src/platform/common/externalinput.cc
@@ -1333,7 +1333,7 @@ HcclResult ParseCannVersion()
         return HCCL_E_NOT_FOUND;
     }
     g_externalInput.cannVersion = std::string(hcommVersion.data()) + "_" + std::string(hcclVersion.data());
-    HCCL_RUN_INFO("[Parse][CannVersion]success, CannVersion is %s ", g_externalInput.cannVersion.c_str());
+    HCCL_RUN_INFO("[Parse][CannVersion]success, hcomm version is %s, hccl version is %s ", std::string(hcommVersion.data()).c_str(),std::string(hcclVersion.data()).c_str());
     return HcclResult::HCCL_SUCCESS;
 #else
 	HCCL_WARNING("[ParseCannVersion]Does not support this interface.");
diff --git a/test/ut/stub/llt_hccl_stub.cc b/test/ut/stub/llt_hccl_stub.cc
index ef2e55365d96e3f782f308f4350035d45c2dd02c..6c3e1fc8414af4177aa5dbead5b7948a1aad2619 100644
--- a/test/ut/stub/llt_hccl_stub.cc
+++ b/test/ut/stub/llt_hccl_stub.cc
@@ -5348,7 +5348,8 @@ const char *aclrtGetSocName()
     return "Ascend910";
 }
 
-ACL_FUNC_VISIBILITY aclError aclsysGetVersionStr(char* pkgNname, char* versionStr) {
+ACL_FUNC_VISIBILITY aclError aclsysGetVersionStr(char* pkgNname, char* versionStr) 
+{
     sal_memcpy(versionStr, sizeof("8.5.0"), "8.5.0", sizeof("8.5.0"));
     return ACL_SUCCESS; 
 }
